
* use the last loaded listener rather than the first to avoid
spamming the SSL error log with dummy connection errors

* don't do a reverse lookup on the bound listener address

* do include the error message for dummy connect failures

Upstream-HEAD: vendor
Upstream-2.0: vendor
Upstream-Status: Specific to the default httpd.conf used.

--- httpd-2.0.51/server/mpm_common.c.pod
+++ httpd-2.0.51/server/mpm_common.c
@@ -384,6 +384,7 @@
 AP_DECLARE(apr_status_t) ap_mpm_pod_open(apr_pool_t *p, ap_pod_t **pod)
 {
     apr_status_t rv;
+    ap_listen_rec *lr;
 
     *pod = apr_palloc(p, sizeof(**pod));
     rv = apr_file_pipe_create(&((*pod)->pod_in), &((*pod)->pod_out), p);
@@ -394,8 +395,11 @@
     apr_file_pipe_timeout_set((*pod)->pod_in, 0);
     (*pod)->p = p;
 
-    apr_sockaddr_info_get(&(*pod)->sa, ap_listeners->bind_addr->hostname,
-                          APR_UNSPEC, ap_listeners->bind_addr->port, 0, p);
+    /* Find the listener specified first in the configuration. */
+    for (lr = ap_listeners; lr->next != NULL; lr = lr->next)
+        /* noop */;
+
+    (*pod)->sa = lr->bind_addr;
 
     /* close these before exec. */
     apr_file_unset_inherit((*pod)->pod_in);
@@ -506,7 +510,7 @@
         }
 
         ap_log_error(APLOG_MARK, log_level, rv, ap_server_conf,
-                     "connect to listener");
+                     "connect to listener on %pI", pod->sa);
     }
 
     apr_socket_close(sock);
